src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/ArchiveAssemblyHelper.cs

```diff
@@ -64,8 +64,57 @@ public ArchiveAssemblyHelper (string archivePath, bool useAssemblyStores = true,
 				ret = ReadZipEntry (path, arch, uncompressIfNecessary);
 			}
 
-			ret?.Seek (0, SeekOrigin.Begin);
-			return ret;
+			if (ret == null) {
+				return null;
+			}
+
+			ret.Flush ();
+			ret.Seek (0, SeekOrigin.Begin);
+			(ulong elfPayloadOffset, ulong elfPayloadSize, ELFPayloadError error) = Xamarin.Android.AssemblyStore.Utils.FindELFPayloadSectionOffsetAndSize (ret);
+
+			if (error != ELFPayloadError.None) {
+				string message = error switch {
+					ELFPayloadError.NotELF           => $"Entry '{path}' is not a valid ELF binary",
+					ELFPayloadError.LoadFailed       => $"Entry '{path}' could not be loaded",
+					ELFPayloadError.NotSharedLibrary => $"Entry '{path}' is not a shared ELF library",
+					ELFPayloadError.NotLittleEndian  => $"Entry '{path}' is not a little-endian ELF image",
+					ELFPayloadError.NoPayloadSection => $"Entry '{path}' does not contain the 'payload' section",
+					_                                => $"Unknown ELF payload section error for entry '{path}': {error}"
+				};
+				Console.WriteLine (message);
+			} else {
+				Console.WriteLine ($"Extracted content from ELF image '{path}'");
+			}
```

Comment:
Should these be going to an MSBuild log instead of `Console`? They won't appear in log files.

Reply:
The only downside is they will be emitting stuff on users machines as well, especially those building on VScode or CLI. And no amount of setting the log verbosity will get rid of them.

Reply:
They are only used in tests, this code doesn't run as part of a regular app build.